uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/TOILRequest.php

```diff
@@ -141,7 +141,7 @@ private static function validateValidTOILAmountNotGreaterThanMaximum($params) {
     $absenceType = AbsenceType::findById($params['type_id']);
     $unlimitedAccrual = empty($absenceType->max_leave_accrual) && $absenceType->max_leave_accrual != 0;
 
-    $currentPeriod = AbsencePeriod::getCurrentPeriod();
+    $currentPeriod = AbsencePeriod::getPeriodContainingDates(new DateTime($params['from_date']), new DateTime($params['to_date']));
```

Comment:
You also need to change the variable name since it might not always be the currentPeriod